src/docs/arch-cursor.dox

```diff
@@ -417,4 +417,33 @@ Thus, in the presence of a memory tracker, uses
 of "stray" pointers will be detected.  Even without a memory tracker, uses of "stray" pointers into
 the freed storage will likely yield the overwritten bytes, and not the previously seen key or value.
 
+@section arch_random_cursor Random Cursor
+
+WiredTiger uses a pseudo-random algorithm when using random cursors with the goal to achieve a good
+random distrubtion and being performant. Only the cursor->next function is used to fetch a random record.
+The current pseudo-random algorithm is as follows:
+
+1. Starting from the root page we descend randomly down using WiredTiger's random number generator (RNG) until we
+reach a leaf page. Once we found a leaf page, we go into different conditions to grab a record from the page. 
+2. First condition: If the page has a sufficiently large number of disk records (1000 or more entries), attempt to find a 
+random visible disk record from the page using WiredTiger's RNG.
+3. Second condition: Loop through all insert lists, and find one that is large (1000 or more entries), once found attempt
+to find a random insert record from the page.
+4. Third condition: If page has small amount of disk records (200 or less entries) or the leaf page is clean, attempt to
+find a random visible disk record from the page using WiredTiger's RNG.
+5. If all conditions fail, compute a random number between 0 - 250 and perform cursor->next or cursor->prev a random
```

Comment:
You say at the start only `next()` can be used

Reply:
Yep! I have already added it here,

> Only the cursor->next function can be used to fetch a random record.

Reply:
But here you mention `prev()` too

Reply:
Hmm I see, the confusion here is that only next() can be used when using the random cursor. But within the random cursor->next() it calls btcur_next or btcur_prev to move between the next record or previous record in the btree. I have reworded it to capture this meaning.